### ISSUE TITLE
datarfame.loc question

### ISSUE THREAD

**Issue by Harshitg** [databricks/koalas#1378]
```
import databricks.koalas as kd
kdf = kd.DataFrame([[0, 2, 3], [0, 4, 1], [10, 20, 30]],index=[4, 5, 5], columns=['A', 'B', 'C'])
```
```
kdf
    A   B   C
4   0   2   3
5   0   4   1
5  10  20  30
```
Operation 
```
kdf['C'].loc[(kdf["A"] <= kdf["B"])] = 10
kdf
    A   B   C
4   0   2   3
5   0   4   1
5  10  20  30
```
The output is incorrect "C" should have become 10.

or Operation 

`kdf.loc[kdf["A"] <= kdf["B"],['C']] = 10
`
```
kdf
SparkPandasNotImplementedError: Can only assign value to the whole dataframe, the row index
                    has to be `slice(None)` or `:` You are trying to use pandas function .loc[..., ...] = ..., use spark function withColumn, select
```

Is there anything that is getting missed?

**Comment by Harshitg** [comment]
Found a workaround as follows:

`kdf['C'] = kdf['C'].mask(kdf["A"] <= kdf["B"],10)
`

**Comment by itholic** [comment]
Let me take a look at this one.

**Comment by HyukjinKwon** [comment]
Thanks, @itholic

**Comment by ueshin** [comment]
FYI: `kdf.loc[kdf["A"] <= kdf["B"], 'C'] = 10` should work now.

**Comment by itholic** [comment]
Anyway, we have same problem with `iLocIndexer`

```python
>>> kdf = ks.DataFrame([[0, 2, 3], [0, 4, 1], [10, 20, 30]],index=[4, 5, 5], columns=['A', 'B', 'C'])
>>> kdf
    A   B   C
4   0   2   3
5   0   4   1
5  10  20  30

>>> kdf["C"].iloc[1:] = 10
>>> kdf
    A   B   C
4   0   2   3
5   0   4   1
5  10  20  30
```

**Comment by ueshin** [comment]
All the issues mentioned above were fixed at #1592.

```py
>>> kdf = ks.DataFrame([[0, 2, 3], [0, 4, 1], [10, 20, 30]],index=[4, 5, 5], columns=['A', 'B', 'C'])
>>> kdf['C'].loc[(kdf["A"] <= kdf["B"])] = 10
>>> kdf
    A   B   C
4   0   2  10
5   0   4  10
5  10  20  10
```
```py
>>> kdf = ks.DataFrame([[0, 2, 3], [0, 4, 1], [10, 20, 30]],index=[4, 5, 5], columns=['A', 'B', 'C'])
>>> kdf.loc[kdf["A"] <= kdf["B"],['C']] = 10
>>> kdf
    A   B   C
4   0   2  10
5   0   4  10
5  10  20  10
```
```py
>>> kdf = ks.DataFrame([[0, 2, 3], [0, 4, 1], [10, 20, 30]],index=[4, 5, 5], columns=['A', 'B', 'C'])
>>> kdf["C"].iloc[1:] = 10
>>> kdf
    A   B   C
4   0   2   3
5   0   4  10
5  10  20  10
```
